### ISSUE TITLE
Doesn't work if cwd is not the git repo root

### ISSUE THREAD

**Issue by okcompute** [wincent/vcs-jump#3]
Hi Wincent,

Thank you very much for this plugin. It is really helpful. But for now, I only get a partial result. I have `autochdir` enabled and it looks to be conflicting with this plugin. If I'm editing a file deep in the project folder tree, I don't see diff for files above. Only if I call the plugin from a buffer referring a file at root that I will see all the git repo diffs.

Is this the intended behaviour? I've looked at the script code but I'm not fluent in Ruby so it was not clear to me if this is a bug or intentional. 

Thanks again!

**Comment by wincent** [comment]
Not so much "intended" behavior as a "expected", simply because I don't use `'autochdir'` and so it's not a use case that I ever really tested. If you look at the code, you can see in there is already a bit of logic to find the repo root and construct paths that are relative to it, but evidently you are running into some untested edge case.

I don't think it should be hard to make it work. Are you in a Git or Mercurial repo? Which commands (`grep`, `diff` or `merge`) are you having problems with?

**Comment by okcompute** [comment]
I'm using both Git and Mercurial repositories. But I found the issue on a Git repo. I'm using both the `merge` and `diff` commands (I use another tool for grepping :) ). I can take a deeper look later to understand what is going and try to fix. 

Thank you for your quick reply :)

**Comment by wincent** [comment]
Ok, let me know what you find out. In a quick test, I can confirm that `diff` works when `cd`-ed into a subdirectory, but if I turn on `'autochdir'` and then open another file, it doesn't work. In fact, `cd`-ing even without `'autochdir'` doesn't work, so it is very likely something being memoized. Actually, looking at it now it is almost certainly [this line](https://github.com/wincent/vcs-jump/blob/369a3fbdab8bb1fdd1b982c9c2bded3a130b66e6/bin/vcs-jump#L42). You can delete the `@pwd ||=` prefix to turn off the memoization, but I don't think that's the permanent fix for this (the memoization is there because that method will get called for every changed file; may or may not be a big deal depending on how many changes you have).

**Comment by okcompute** [comment]
I pushed a tentative PR (#4) to fix the issue.

**Comment by wincent** [comment]
Give https://github.com/wincent/vcs-jump/commit/5016306e8c7f73a45cd6d1e7c65ee067ddeea15e a try; I think that it might work for your use case — let me know if it doesn't.

**Comment by okcompute** [comment]
Thank you so much! Sadly, it still doesn't work for the use case where I launch Vim outside of a repository. Let say I have two repositories, one in folder /A and the other in folder /B. If I open Vim in folder /A and then open a file from folder /B, `VcsJump diff` won't work. I switched back and forth between VcsJump`master` branch and my _flawed_ tentative branch. `master` does not work, my branch does. 

I was wondering, could we simply fix this issue by just cd-ing in the current buffer `pwd`?

i.e. :

```
  cexpr system('cd ' . shellescape(expand("%:p:h")). ' && ' .   s:jump_path . ' ' . l:command . ' ' . ' 2> /dev/null')

```

This way, we keep the ruby script API intact! What do you think?

Again, thank you very much for taking the time to look at this issue!

**Comment by okcompute** [comment]
Just tried and looks like my proposal above ☝️ works!  :)

**Comment by okcompute** [comment]
Shorter version:

`  cexpr system('cd '. expand("%:p:h:S") . ' && ' . s:jump_path . ' ' . l:command . ' 2> /dev/null')
`

**Comment by wincent** [comment]
> I was wondering, could we simply fix this issue by just cd-ing in the current buffer

Yes, this was what I suggested [over here](https://github.com/wincent/vcs-jump/pull/4#pullrequestreview-294692931), however, I was thinking that it might not be needed after the changes that I pushed, because, in theory, if you have `'autochdir'` set, isn't the `:pwd` when you open a file from folder `/B` already set to that file's folder? What does `:pwd` print when you're looking at that file? And what does `:echo system('pwd')` print?

**Comment by okcompute** [comment]
(Sorry for my late response)

> Yes, this was what I suggested over here,

I'm sorry. At the time I did not realize how simple it would have been to make this change!

> in theory, if you have 'autochdir' set, isn't the :pwd set, isn't the :pwd when you open a file from folder /B already set to that file's folder?

I forgot to mention something. Since dabbling with this issue, I realized other plugins have issues with `autochdir` (even the attribute documentation warns about potential plugin issues!). I disabled the `autochdir` to see if I could live without it. So, in that context, if you open a file somewhere outside the current `pwd`, we end up with the same original problem. Though, the simple fix I propose ☝️ works in all case. Should I push a PR?

Thank you!
